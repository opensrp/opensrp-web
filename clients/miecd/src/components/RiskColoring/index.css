.green {
  background-color: rgba(152, 255, 131, 0.24);
  border-radius: 10px;
  border: 2px solid rgba(152, 255, 131) !important;
}

.blue {
  background-color: rgba(138, 228, 255, 0.24);
  border-radius: 10px;
  border: 2px solid rgba(138, 228, 255) !important;
}

.yellow {
  background-color: rgba(252, 255, 104, 0.24);
  border-radius: 10px;
  border: 2px solid rgba(252, 255, 104) !important;
}

.red {
  background-color: rgba(255, 196, 196, 0.24);
  border-radius: 10px;
  border: 2px solid rgba(255, 196, 196) !important;
}

.risk-default {
  background-color: None;
}

.badge {
  padding: 0;
  width: 70px;
  height: 24px;
  border-radius: 1.25rem;
}

.badge div {
  height: 100%;
  position: relative;
}

.badge div p {
  text-align: center;
  position: relative;
  font-size: 12px;
  font-weight: 400;
  line-height: 24px;
}

.badge-danger {
  background-color: #d0021b;
}

.badge-warning {
  background-color: #ff8533;
}

.badge-info {
  background-color: inherit;
  color: #9596a3;
}
