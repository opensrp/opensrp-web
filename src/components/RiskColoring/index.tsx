import React from 'react';
import { HIGH_RISK, LOW_RISK, NO_RISK, NOT_SET, RED_ALERT } from '../../constants';
import './index.css';

interface RiskColoringProps {
  Risk: string;
}

const riskColoringDefaultProps: RiskColoringProps = {
  Risk: 'red-alert',
};

/** Colour codes risk levels */
const riskColoring = (props: RiskColoringProps) => {
  const { Risk } = props;
  switch (Risk.toLowerCase()) {
    case 'no risk':
      return (
        <span className="badge badge-success" id="default-width">
          <div>
            <p>{NO_RISK}</p>
          </div>
        </span>
      );
    case 'low':
      return (
        <span className="badge badge-primary" id="default-width">
          <div>
            <p>{LOW_RISK}</p>
          </div>
        </span>
      );
    case 'high':
      return (
        <span className="badge badge-warning" id="default-width">
          <div>
            <p>{HIGH_RISK}</p>
          </div>
        </span>
      );
    case 'red':
      return (
        <span className="badge badge-danger" id="default-width">
          <div>
            <p>{RED_ALERT}</p>
          </div>
        </span>
      );
    case 'not set':
      return (
        <span className="badge badge-info" id="default-width">
          <div>
            <p>{NOT_SET}</p>
          </div>
        </span>
      );
    default:
      return (
        <span className="badge badge-info" id="default-width">
          {Risk}
        </span>
      );
  }
};

riskColoring.defaultProps = riskColoringDefaultProps;

export default riskColoring;
