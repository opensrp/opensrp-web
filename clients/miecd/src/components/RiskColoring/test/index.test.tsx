import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import RiskColoring from '../index';

describe('RiskColoring', () => {
    it('must render without crashing', () => {
        shallow(<RiskColoring />);
    });

    it('must show correct Risk level depending on the string passed as a prop', () => {
        let wrapper = mount(<RiskColoring risk={'high'} />);
        expect(toJson(wrapper)).toMatchSnapshot('hight_risk');
        wrapper = mount(<RiskColoring risk={'low'} />);
        expect(toJson(wrapper)).toMatchSnapshot('low_risk');
        wrapper = mount(<RiskColoring risk={'red'} />);
        expect(toJson(wrapper)).toMatchSnapshot('red_alert');
        wrapper = mount(<RiskColoring risk={'not set'} />);
        expect(toJson(wrapper)).toMatchSnapshot('not_set');
        wrapper = mount(<RiskColoring />);
        expect(toJson(wrapper)).toMatchSnapshot('default');
    });
});
