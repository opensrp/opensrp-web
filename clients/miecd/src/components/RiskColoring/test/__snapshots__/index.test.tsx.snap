// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RiskColoring must show correct Risk level depending on the string passed as a prop: default 1`] = `
<riskColoring
  risk="red-alert"
>
  <span
    className="badge badge-info"
    id="default-width"
  >
    red-alert
  </span>
</riskColoring>
`;

exports[`RiskColoring must show correct Risk level depending on the string passed as a prop: hight_risk 1`] = `
<riskColoring
  risk="high"
>
  <span
    className="badge badge-danger"
    id="default-width"
  >
    <div>
      <p>
        High Risk
      </p>
    </div>
  </span>
</riskColoring>
`;

exports[`RiskColoring must show correct Risk level depending on the string passed as a prop: low_risk 1`] = `
<riskColoring
  risk="low"
>
  <span
    className="badge badge-warning"
    id="default-width"
  >
    <div>
      <p>
        Low Risk
      </p>
    </div>
  </span>
</riskColoring>
`;

exports[`RiskColoring must show correct Risk level depending on the string passed as a prop: not_set 1`] = `
<riskColoring
  risk="not set"
>
  <span
    className="badge badge-info"
    id="default-width"
  >
    <div>
      <p>
        Not Set
      </p>
    </div>
  </span>
</riskColoring>
`;

exports[`RiskColoring must show correct Risk level depending on the string passed as a prop: red_alert 1`] = `
<riskColoring
  risk="red"
>
  <span
    className="badge badge-danger"
    id="default-width"
  >
    <div>
      <p>
        Red Alert
      </p>
    </div>
  </span>
</riskColoring>
`;
